internal/controller/deployment_to_pdb_controller.go

```diff
@@ -65,6 +68,36 @@ func (r *DeploymentToPDBReconciler) handleDeploymentReconcile(ctx context.Contex
 
 			// PDB already exists, nothing to do
 			log.Info("PodDisruptionBudget already exists", "namespace", pdb.Namespace, "name", pdb.Name)
+			pdbWatcher := &myappsv1.PDBWatcher{}
+			e := r.Get(ctx, types.NamespacedName{Name: pdb.Name, Namespace: pdb.Namespace}, pdbWatcher)
+			if e == nil {
+				// if pdb exists get pdbWatcher --> compare targetGeneration field for deployment if both not same deployment was not changed by pdb watcher
+				// update pdb minReplicas to current deployment replicas
+				log.Info("deployment replicas got updated", " pdbWatcher.Status.TargetGeneration", pdbWatcher.Status.TargetGeneration, "deployment.Generation", deployment.GetGeneration())
+				if pdbWatcher.Status.TargetGeneration != deployment.GetGeneration() {
```

Comment:
Think I had a comment about this before but pdbwatcher updates deployment then itsself in a non atomic way. So you could actually end up here before pdbwatcher.staus.targetgenration is changed. 

You at very least need comment here (pdb watcher also needs comment saying it fails between updating deployment and targetgeneration it can leak) but maybe better would be to wait and redo this reoncile a couple of second later to try and avoid that race?

Reply:
Ah okay but the annotations are atomic. thats more fun. Can you leave a comment about how the annotations avoid race conditon?